libraries/cmake/README.md

```diff
@@ -80,7 +80,58 @@ uname -r
 4.15.0-1099-aws
 ```
 
-### Troubleshooting Linux
+### Troubleshooting CentOS 6 Linux
+
+CentOS 6 reached "End of Life" status in 2020, so continuing to build osquery on it requires some extra preparation steps.
+
+#### Yum Package Repo
+
+The Yum package repo for CentOS 6 is no longer hosted at its default location, so we must configure it.
+
+```sh
+sudo curl https://www.getpagespeed.com/files/centos6-eol.repo --output /etc/yum.repos.d/CentOS-Base.repo
+sudo yum update
+```
+
+#### Upgrading git
+
+The version of `git` on CentOS 6 is ~1.7, but osquery requires much newer. We can install it from source.
+
+```sh
+sudo yum remove git
+sudo yum -y install curl-devel expat-devel gettext-devel openssl-devel zlib-devel gcc perl-ExtUtils-MakeMaker
+cd /usr/src
+sudo wget https://www.kernel.org/pub/software/scm/git/git-2.39.0.tar.gz
+sudo tar xzf git-2.39.0.tar.gz
+cd git-2.39.0
+sudo make prefix=/usr/local all
+sudo make prefix=/usr/local install
+sudo ln -sfn /usr/local/bin/git /usr/bin/git
+```
+
+#### Upgrading Python
+
+The version of `Python` on CentOS 6 is too old to complete osquery's CMake configuration steps. We can install Python 3 from source as well.
+
+```sh
+sudo yum -y install gcc openssl-devel bzip2-devel
+wget https://www.python.org/ftp/python/3.6.6/Python-3.6.6.tgz
+tar xzf Python-3.6.6.tgz
+cd Python-3.6.6
+./configure --enable-optimizations
+sudo make altinstall
+sudo ln -sfn /usr/local/bin/python3.6 /usr/bin/python3
+curl -O https://bootstrap.pypa.io/pip/3.6/get-pip.py
+sudo python3 get-pip.py
+```
```

Comment:
For python there is the SCL repository. We have some instructions on the dpkg README that we could move here, instead of building it ourselves: https://github.com/osquery/osquery/tree/master/libraries/cmake/source/libdpkg#install-python-36-x86-only